src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+        [Theory]
+        [MemberData("RemoveAtInvalidData")]
+        public void RemoveAtInvalidReadOnly(int index, int size)
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(size);
+            var list = GetList(items);
+            AssertThrows(new []{typeof(ArgumentOutOfRangeException), typeof(NotSupportedException)}, () => list.RemoveAt(index));
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void RemoveAtFixedSizeThrows()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+            {
+                var items = GenerateItems(16);
+                var list = GetList(items);
+                Assert.Throws<NotSupportedException>(
+                    () => list.RemoveAt(0));
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Theory]
+        [InlineData(32)]
+        [InlineData(16)]
+        public void RemoveAtDescending(int size)
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(size);
+            var list = GetList(items);
+            for (int i = 0; i < size; i++)
+            {
+                list.RemoveAt(size - i - 1);
+                items = items.Slice(0, items.Length - 1);
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Theory]
+        [InlineData(32)]
+        [InlineData(16)]
+        public void RemoveAt(int size)
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(size);
+            var list = GetList(items);
+
+            for (int i = 0; i < size; i++)
+            {
+                list.RemoveAt(0);
+                items = items.Slice(1);
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Theory]
+        [InlineData(32)]
+        [InlineData(16)]
+        public void RemoveAtEachIndex(int size)
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(size);
+            var list = GetList(items);
+            for (int i = 0; i < size; i++)
+            {
+                list.RemoveAt(i);
+                CollectionAssert.Equal(items.RemoveAt(i), list);
+                list.Insert(i, items[i]);
+            }
+        }
+
+        /// <summary>
+        ///     When overridden in a derived class, Gets a list of values that are not valid elements in the collection under test.
+        /// </summary>
+        /// <returns>An <see cref="IEnumerable"/> containing the invalid values.</returns>
+        protected abstract IEnumerable GetInvalidValues();
+
+        /// <summary>
+        ///     When overridden in a derived class, Gets an instance of the list under test containing the given items.
```

Comment:
The comment doesn't match the method.  (The comment is for a virtual, this is for a non-virtual (private, even)).